### PR TITLE
Add utils and services for functions

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes

- Add authentication, logging, API utils
- Add unit tests
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: services/__tests__/storage.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const blobClientStub = {
+    getContainerClient: (_: any) => containerClientStub
+  };
+
+  const res = await listBlobs(blobClientStub as any, BLOB_NAME).next();
+
+  expect(res.value).toBe(expectedBlob)
+});
+
+it('It should sucessfully featch a blob', async () => {
```
Comment (infantesimone) [inline services/__tests__/storage.test.ts:46]: can you differentiate the description from the previous test?

File: services/__tests__/storage.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { BlobItem } from '@azure/storage-blob';
```
Comment (infantesimone) [inline services/__tests__/storage.test.ts:1]: To add test when getLastBlob => undefined

File: services/storage.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  return containerClient.listBlobsFlat();
+};
+
+export const getLastBlob = async (
+  blobServiceCliet: BlobServiceClient,
+  container: string
+): Promise<BlobItem | undefined> => {
+  let res;
+
+  for await (const blob of listBlobs(blobServiceCliet, container)) {
```
Comment (infantesimone) [inline services/storage.ts:20]: sounds good, but we could create a blob naming convention based on a timestamp, to avoid `for`. As an improvement to do in another PR  (if you are agree cc @MarcoAGreco )

**Review by infantesimone — APPROVED** [review]
few small comments, but PR sounds good  🚀🚀🚀

File: services/__tests__/storage.test.ts (hunk 1 above)
Comment (MarcoAGreco) [reply services/__tests__/storage.test.ts:46]: My fault, I'll add a correct description 👍

File: services/storage.ts (hunk 3 above)
Comment (MarcoAGreco) [reply services/storage.ts:20]: I agree, it seems a better solution 👍
